pkg/noderesourcetopology/least_numa.go

```diff
@@ -153,3 +167,78 @@ func combineResources(numaNodes NUMANodeList, combination []int) v1.ResourceList
 
 	return resources
 }
+
+// numaNodesRequired returns bitmask with minimal NUMA nodes required to run given resources
+// or nil when resources can't be fitted onto the worker node
+// second value returned is a boolean indicating if bitmask is optimal from distance perspective
+func numaNodesRequired(identifier string, qos v1.PodQOSClass, numaNodes NUMANodeList, resources v1.ResourceList) (bitmask.BitMask, bool) {
+	for bitmaskLen := 1; bitmaskLen <= len(numaNodes); bitmaskLen++ {
+		numaNodesCombination := combin.Combinations(len(numaNodes), bitmaskLen)
+		suitableCombination, isMinDistance := findSuitableCombination(identifier, qos, numaNodes, resources, numaNodesCombination)
+		// we have found suitable combination for given bitmaskLen
+		if suitableCombination != nil {
+			bm := bitmask.NewEmptyBitMask()
+			bm.Add(suitableCombination...)
+			return bm, isMinDistance
+		}
+	}
+
+	return nil, false
+}
+
+func onlyNonNUMAResources(numaNodes NUMANodeList, resources v1.ResourceList) bool {
```

Comment:
from past experience, this can be surprisingly trick to get right. Let's move into `pluginhelpers.go` and let's add unit tests. If you look in git history, you should find similar code we added and then removed in the past, that could help you

Reply:
:+1:

Reply:
I have added unit tests for that.